tests/tests_pytorch/trainer/optimization/test_manual_optimization.py

```diff
@@ -304,8 +304,28 @@ def on_train_epoch_end(self, *_, **__):
     trainer.fit(model)
 
 
+class CustomMapping(collections.abc.Mapping):
+    """A custom implementation of Mapping for testing purposes."""
+
+    def __init__(self, *args, **kwargs):
+        self._store = dict(*args, **kwargs)
+
+    def __getitem__(self, key):
+        return self._store[key]
+
+    def __iter__(self):
+        return iter(self._store)
+
+    def __len__(self):
+        return len(self._store)
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self._store})"
+
```

Comment:
```suggestion
    def __repr__(self):
        return f"{self.__class__.__name__}({self._store})"

    def __copy__(self):
        cls = self.__class__
        new_obj = cls(self._store.copy())
        return new_obj
```

Seems like we need a copy method here to make sure copy works in [`.../manual.py`](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/loops/optimization/manual.py#L49)